### PR TITLE
memory limit and overflow checking

### PR DESCRIPTION
**Pull request by smaznet**

### CODE REVIEW

**Review by vrumger — CHANGES_REQUESTED** [review]
Please use `snakeCase` for variable names.

**Comment by rojserbest** [comment]
@AndrewLaneX it is fixed

File: src/stream.ts [hunk 1]
```diff
@@ -30,6 +37,14 @@ export class Stream extends EventEmitter {
...
     readonly channelCount: number;
     private almostFinishedTrigger: number;
+    private byteLength: number;
+    private cacheSize: number = 0;
+    private playedBytes = 0;
+    private chunk: Buffer;
+    private _readablePaused = false;
+    remoteTime?: RemotePlayingTimeCallback;
+    remoteLagging?: RemoteLaggingCallback;
+    overflowCallback?: (pause: boolean) => void;
```
Comment (vrumger) [inline src/stream.ts:47]: What's this for?

Comment (smaznet) [reply src/stream.ts]: some of codes was from @Laky-64 project
but in my implementation i didn't use that variables

original source: https://github.com/pytgcalls/pytgcalls/blob/master/src/stream.ts

**Review by vrumger — APPROVED** [review]
Thanks!

**Pull request merged**